hedged.go

```diff
@@ -12,6 +12,79 @@ import (
 
 const infiniteTimeout = 30 * 24 * time.Hour // domain specific infinite
 
+// Client represents a hedged HTTP client.
+type Client struct {
+	rt    http.RoundTripper
+	stats *Stats
+}
+
+// Config for the [Client].
+type Config struct {
+	// Transport of the [Client].
+	// Default is nil which results in [net/http.DefaultTransport].
+	Transport http.RoundTripper
+
+	// Upto says how much requests to make.
+	// Default is zero which means no hedged requests will be made.
+	Upto int
+
+	// Delay before 2 consequitive hedged requests.
+	Delay time.Duration
+
+	// Next returns the upto and delay for each HTTP that will be hedged.
+	// Default is nil which results in (Upto, Delay) result.
+	Next NextFn
+}
+
+// NextFn represents a function that is called for each HTTP request.
```

Comment:
```suggestion
// HedgeStrategyFn represents a function that is called for each HTTP request for retrieving hedging options dynamically.
```
Maybe something like this sounds better? I'm not sure what `Next` refers to

Reply:
`Next` is a popular name for iterator-like patterns. This also reminds me a backoff implementation https://github.com/cristalhq/synx/blob/main/backoff.go#L12

I prefer to leave `Next` 'cause it's shorter and used in only 1 place basically.